src/test/java/cloud/fogbow/accs/core/datastore/services/RecordServiceTest.java

```diff
@@ -0,0 +1,156 @@
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.sql.Timestamp;
+import java.text.ParseException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@PowerMockIgnore({"javax.management.*"})
+@PrepareForTest({RecordRepository.class, RecordService.class})
+@RunWith(PowerMockRunner.class)
+@PowerMockRunnerDelegate(SpringRunner.class)
+@SpringBootTest
+public class RecordServiceTest extends ServicesBaseUnitTest{
+
+    private final String ANY_VALUE = "any";
+    private final String TEST_DATE = "2000-01-01";
+    private final String DEFAULT_RESOURCE_TYPE = "compute";
+
+    //test case: When the start time is greater than the end time the
+    //message checked below must be part of the threw exception.
+    @Test
+    public void testCheckIntervalWhenStartTimeIsGreaterThanAllowed() {
+        //setup
+        Timestamp endTimeStamp = new Timestamp(System.currentTimeMillis());
+        Timestamp startTimeStamp = new Timestamp(System.currentTimeMillis() + 1);
+
+        try {
+            //exercise
+            recordService.checkInterval(startTimeStamp, endTimeStamp);
+        } catch (InvalidIntervalException ex) {
+            //verify
+            Assert.assertEquals(Messages.Exception.START_TIME_GREATER_THAN_END_TIME, ex.getMessage());
+        }
+    }
+
+    //test case: When the start time is greater than now the
+    //message checked below must be part of the threw exception.
+    @Test
+    public void testCheckIntervalWhenBillingPredictions() {
+        //setup
+        Timestamp startTimestamp = new Timestamp(System.currentTimeMillis() + 60000*2);
+        Timestamp endTimestamp = new Timestamp(System.currentTimeMillis() + 60000*3);
+
+        try {
+            //exercise
+            recordService.checkInterval(startTimestamp, endTimestamp);
+        } catch(InvalidIntervalException ex) {
+            //verify
+            Assert.assertEquals(Messages.Exception.BILLING_PREDICTIONS, ex.getMessage());
+        }
+    }
+
+    //test case: check if the timestamp returned is the expected one.
+    @Test
+    public void testGetTimestampFromString() throws ParseException{
+        //setup
+        Timestamp twoThousandTimestamp = Timestamp.valueOf("2000-01-01 00:00:00");
+
+        //exercise/verify
+        Assert.assertEquals(twoThousandTimestamp, recordService.getTimestampFromString(TEST_DATE));
+    }
+
+    //test case: just exercise the method by checking if it returned what it should return
+    //(exactly what the repository returned, once its up to spring to test the correctness of the repository).
+    @Test
+    public void testGetOpenedRecords() {
+        //setup
+        mockDatabaseOperations(RECORDS_BY_STATE, OrderState.FULFILLED, null, null, 2);
+
+        //exercise
+        List<Record> records = recordService.getOpenedRecords(getAccountingUser(),"mockedRequestingMember", DEFAULT_RESOURCE_TYPE,
```

Comment:
Transform the string "mockedRequestingMember" into a constant of this class and use it between test methods.